EpikCord/__init__.py

```diff
@@ -1156,6 +1113,151 @@ async def command_error(
     ):
         logger.exception(error)
 
+    #AutoModeration Events
+    async def auto_moderation_rule_create(self, data:dict) -> AutoModerationRule:
+        rule = AutoModerationRule(self, data)
+        self.auto_moderation_cache.add_to_cache(rule.guild_id, rule)
+        return rule
+    
+    async def auto_moderation_rule_update(self, data:dict)->tuple[Optional[AutoModerationRule], AutoModerationRule]:
+        after:AutoModerationRule = AutoModerationRule(self,data)
+        before:Optional[AutoModerationRule] = self.auto_moderation_cache.get(after.guild_id)
+        return before,after
+        
+    async def auto_moderation_rule_delete(self, data:dict)->Optional[AutoModerationRule]:
+        rule_guild_id = data["guild_id"]
+        previous:AutoModerationRule = self.auto_moderation_cache.get(rule_guild_id)
+        return previous
+
+    async def auto_moderation_action_execution(self, data:dict) -> AutoModerationExecutionEvent:
+        return AutoModerationExecutionEvent(data)    
+
+    # Guild Events
+    async def guild_create(self, data) -> Union[UnavailableGuild, Guild]:
+        guild = (
+            UnavailableGuild(data)
+            if data.get("unavailable") is True
+            else Guild(self, data)
+            if data.get("unavailable") is False
+            else None
```

Comment:
no

Reply:
I did not do that

Reply:
Ask @TheUntraceable

Reply:
For some reason they just shove all Guilds into a single event, by all guilds I mean Unavailable ones and Available ones (Guilds), and so I used that to determine if the guild is new.

Reply:
This code needs to be refactor, chaining ternary is a bad practice and should be avoided, it make the code hard to read